extensions/ql-vscode/src/model-editor/model-evaluator.ts

```diff
@@ -28,18 +38,48 @@ export class ModelEvaluator extends DisposableObject {
     };
     this.modelingStore.updateModelEvaluationRun(this.dbItem, evaluationRun);
 
-    // For now, just wait 5 seconds and then update the store.
-    // In the future, this will be replaced with the actual evaluation process.
-    void sleep(5000).then(() => {
-      const completedEvaluationRun: ModelEvaluationRun = {
-        isPreparing: false,
-        variantAnalysisId: undefined,
-      };
-      this.modelingStore.updateModelEvaluationRun(
-        this.dbItem,
-        completedEvaluationRun,
-      );
-    });
+    // Build pack
+    const qlPack = await resolveCodeScanningQueryPack(
+      this.logger,
+      this.cliServer,
+      this.language,
+    );
+
+    if (!qlPack) {
+      this.modelingStore.updateModelEvaluationRun(this.dbItem, undefined);
+      throw new Error("Unable to trigger evaluation run");
+    }
+
+    // Submit varfiant analysis and monitor progress
+    return withProgress(
+      async (progress, token) => {
+        let variantAnalysisId: number | undefined = undefined;
+        try {
+          variantAnalysisId =
+            await this.variantAnalysisManager.runVariantAnalysis(
+              qlPack,
+              progress,
+              token,
+            );
+        } catch (e) {
+          this.modelingStore.updateModelEvaluationRun(this.dbItem, undefined);
+          throw e;
+        }
```

Comment:
Maybe this is just because things are broken for now, but I'm wondering if we're catching errors properly? Bundling the pack fails (as expected), but gives an "Unhandled error" and then the spinner is stuck on "Stop evaluation".

I think I'd expect the error to occur, but then the button to change back to "Evaluate" 🤔 

![image](https://github.com/github/vscode-codeql/assets/42641846/dd7301d9-4203-42df-9836-045e13ec0dfb)

Reply:
Ah thanks for spotting that, I thought I had fixed it. Fixed with https://github.com/github/vscode-codeql/pull/3397/commits/bf3620b708ffe3add41e761023161a4b85c2975c